lib/driver-cli.js

```diff
@@ -0,0 +1,250 @@
+import _ from 'lodash';
+import path from 'path';
+import { KNOWN_DRIVERS, validateDriversYaml, writeInstalledDriver,
+         INSTALL_TYPE_LOCAL, INSTALL_TYPE_NPM, INSTALL_TYPE_GIT,
+         INSTALL_TYPE_GITHUB } from './drivers';
+import { getLatestVersion, installPackage, linkPackage } from './npm';
+import { errAndQuit, log, spinWith } from './cli-utils';
+import { util } from 'appium-support';
+
+let INSTALLED_DRIVERS;
+
+async function runDriverCommand (args) {
+  // TODO driver.json file should be locked while any of these commands are
+  // running to prevent weird situations
+  let jsonResult = null;
+  const {json, appiumHome} = args;
+  const logFn = (msg) => log(json, msg);
+  try {
+    INSTALLED_DRIVERS = await validateDriversYaml(appiumHome, logFn);
+    switch (args.driverCommand) {
+      case 'list':
+        jsonResult = await list(args);
+        break;
+      case 'install':
+        jsonResult = await install(args);
+        break;
+      case 'uninstall':
+        jsonResult = await uninstall(args);
+        break;
+      case 'update':
+        jsonResult = await update(args);
+        break;
+      case 'verify':
+        // TODO create a 'verify' command that just checks driver.json and then
+        // that everything in driver.json can be required OK.
+        // verify could also have a --reinstall flag which installs any drivers
+        // which are listed in the drivers.json but which cannot be required
+        // for whatever reason
+        break;
+      default:
+        throw new Error(`Cannot handle driver command ${args.driverCommand}`);
+    }
+  } catch (err) {
+    errAndQuit(json, err);
+  }
+
+  if (json) {
+    console.log(JSON.stringify(jsonResult)); // eslint-disable-line no-console
+  }
+}
+
+async function list ({appiumHome, json, showInstalled, showUpdates}) {
```

Comment:
it makes sense to have the commands as members of the same class, since they share common args